### ISSUE TITLE
Possible solution to read only filesystem errors when running build.rs

### ISSUE THREAD

**Issue by TroyNeubauer** [twistedfall/opencv-rust#412]
Hi! Thanks for building such a terrific library. I use it almost every week. 

I also have been running into issues #313 and #299. I have a fork at work which fixes these issues by using a combination of temporary directories and cargo's `OUT_DIR`. I was wondering if you would be interested in looking at a PR that fixes these issues. 

I know that committing the generated files is important for not rerunning builds and IDE quality of life, so I propose that we add an (off by default) feature flag like `read_only_build` (or a better name) in addition to changing `src/opencv/mod.rs` from:
```
pub mod hub;
```
to
```
#[cfg(not(feature(read_only_build)))]
pub mod hub;
#[cfg(feature(read_only_build))]
include!(concat!(env!("OUT_DIR"), "/hub.rs"))
```

`read_only_build` would be off by default and cause no changes for existing users, however it can be opted into to prevent the issues linked above by generating the files into `OUT_DIR`, which is always writable, even if the source directory is not.

**Comment by twistedfall** [comment]
That is the way I've chosen when trying to implement this feature, but it wasn't trivial anyway. I'll try to revive that branch, but one other solution would be to just switch to that by default (and handle docs.rs somehow differently) because IDE's have since become much better in picking up generated sources so that might not be a significant problem any more.

**Comment by twistedfall** [comment]
This should now be fixed in 0.76.0, the bindings are now generated in the `OUT_DIR`.
